include/native/reduce.h

```diff
@@ -25,154 +25,65 @@
 #include <functional>
 
 namespace grppi{
-//typename std::enable_if<!is_iterator<Output>::value, bool>::type,
-
-template < typename InputIt, typename Output, typename ReduceOperator>
- typename std::enable_if<!is_iterator<Output>::value, void>::type 
-reduce(parallel_execution_native &p, InputIt first, InputIt last, Output & firstOut, ReduceOperator op) {
-
-    typename ReduceOperator::result_type identityVal = !op(false,true);
 
+template < typename InputIt, typename ReduceOperator>
+typename std::iterator_traits<InputIt>::value_type
+reduce(parallel_execution_native &p, InputIt first, InputIt last, typename std::iterator_traits<InputIt>::value_type init, ReduceOperator && op){
```

Comment:
In all grppi::reduce the operation type should be renamed to `Combiner` and the argument to `combine_op`.

In all grppi::map_reduce the operation type should be renamed to `Transformer` and the argument to `transform_op`.